### PR TITLE
[ML-10524] Add example GPU containers

### PR DESCRIPTION
**Pull request by mengxr**
Add example GPU containers. Work in based on #19.

* `gpu-base` extends [`minimal`](https://github.com/databricks/containers/tree/master/ubuntu/minimal) with CUDA Toolkit and common GPU libraries.
* `gpu-python` installs common python packages to start with for many Databricks users.
* `gpu-tensorflow` installs TensorFlow onto the `gpu-python` image.
* `gpu-pytorch` installs PyTorch onto the `gpu-python` image.
* `gpu-rapids` installs [RAPIDS](https://rapids.ai/) onto the `gpu-python` image.I published docker files in my own dockerhub. Example cluster JSON for RAPIDS:

~~~json
{
    "cluster_name": "meng-dcs-rapids",
    "spark_version": "7.0.x-scala2.12",
    "node_type_id": "p3.2xlarge",
    "num_workers": 0,
    "docker_image": {
      "url": "mengxr/gpu-rapids:cuda10.1"
    },
    "aws_attributes": {
      "ebs_volume_count": 1,
      "ebs_volume_size": 100,
      "zone_id": "us-west-2b"
    }
}
~~~

Closes #19

### CODE REVIEW

File: ubuntu/gpu/python/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    # Source conda.sh for all login and interactive shells.
+    ln -s /databricks/conda/etc/profile.d/conda.sh /etc/profile.d/conda.sh && \
+    echo ". /etc/profile.d/conda.sh" >> ~/.bashrc && \
+    # Set always_yes for non-interactive shells.
+    conda config --system --set always_yes True
+
+COPY gpu.yml /tmp/gpu.yml
+
+RUN conda env create --file /tmp/gpu.yml && \
+    rm -f /tmp/gpu.yml
```
Comment (evanye) [inline ubuntu/gpu/python/Dockerfile:23]: does `conda env create` modify the file? Why do we need a /tmp/gpu.yml?

File: ubuntu/gpu/python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,26 @@
+FROM databricksruntime/gpu-base:cuda10.1
```
Comment (evanye) [inline ubuntu/gpu/python/Dockerfile:1]: Were you able to build this image? Right now we tag them `latest` for lack of a better tagging/versioning system

File: ubuntu/gpu/rapids/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,9 @@
+FROM databricksruntime/gpu-python:cuda10.1
+
+COPY rapids-spec.txt /tmp/rapids-spec.txt
+
+RUN conda create --name rapids --file /tmp/rapids-spec.txt && \
+    rm -f /tmp/rapids-spec.txt
+
+# Set an environment variable used by Databricks to decide which conda environment to activate by default.
+ENV DEFAULT_DATABRICKS_ROOT_CONDA_ENV=rapids
```
Comment (evanye) [inline ubuntu/gpu/rapids/Dockerfile:9]: This still leaves the old `gpu` conda environment around, right? Should we remove it to save on image size? Probably not worth it for now.

File: ubuntu/gpu/python/Dockerfile (hunk 2 above)
Comment (mengxr) [reply ubuntu/gpu/python/Dockerfile:1]: I tagged it locally to build this image. For GPU containers, the CUDA version is the most important one.

File: ubuntu/gpu/python/Dockerfile (hunk 1 above)
Comment (mengxr) [reply ubuntu/gpu/python/Dockerfile:23]: No. I need to copy the env spec file to somewhere in the docker build. What is your recommendation?

File: ubuntu/gpu/rapids/Dockerfile (hunk 3 above)
Comment (mengxr) [reply ubuntu/gpu/rapids/Dockerfile:9]: Does it really save the image size? The layers are still there. I don't expect users use those containers directly. Those are just examples.

**Pull request merged**